repository/src/main/java/org/alfresco/repo/node/sizedetails/NodeSizeDetailsServiceImpl.java

```diff
@@ -0,0 +1,413 @@
+
+            searchParameters.addFacetQuery(FACET_QUERY);
+            FieldFacet fieldFacet = new FieldFacet(FIELD_FACET);
+            fieldFacet.setLimitOrNull((int) resultsWithoutFacet.getNumberFound());
+            searchParameters.addFieldFacet(fieldFacet);
+            resultsWithoutFacet.close();
+            return searchService.query(searchParameters);
+        }
+        catch (Exception e)
+        {
+            LOG.error("Exception occurred while executing facetQuery ", e);
+            throw e;
+        }
+    }
+
+    private SearchParameters createSearchParameters(NodeRef nodeRef)
+    {
+        SearchParameters searchParameters = new SearchParameters();
+        searchParameters.addStore(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE);
+        searchParameters.setLanguage(SearchService.LANGUAGE_FTS_ALFRESCO);
+        searchParameters.setQuery("ANCESTOR:\"" + nodeRef + "\" AND TYPE:content");
+        searchParameters.setTrackTotalHits(-1);
+        return searchParameters;
+    }
+
+    @Override
+    public void afterPropertiesSet() throws Exception
+    {
+        ParameterCheck.mandatory("searchService", this.searchService);
+        ParameterCheck.mandatory("simpleCache", this.simpleCache);
+        ParameterCheck.mandatory("transactionService", this.transactionService);
+        ParameterCheck.mandatory("threadPoolExecutor", this.threadPoolExecutor);
+    }
+
+    /**
+     * POJO class to hold node size details.
+     */
+    public static class NodeSizeDetails implements Serializable
+    {
+        private static final long serialVersionUID = 1L;
+        private String id;
+        private Long sizeInBytes;
+        private Date calculatedAt;
+        private Integer numberOfFiles;
+        private String jobId;
+        private STATUS status;
+
+        public NodeSizeDetails(String jobId)
+        {
+            this.jobId = jobId;
+        }
+
+        public NodeSizeDetails(String id, STATUS status)
+        {
+            this.id = id;
+            this.status = status;
+        }
+
+        public NodeSizeDetails(String id, String jobId, STATUS status)
+        {
+            this.id = id;
+            this.jobId = jobId;
+            this.status = status;
+        }
+
+        public NodeSizeDetails(String id, Long sizeInBytes, String jobId, STATUS status)
+        {
+            this.id = id;
+            this.sizeInBytes = sizeInBytes;
+            this.jobId = jobId;
+            this.status = status;
+        }
+
+        public NodeSizeDetails(String id, Long sizeInBytes, Date calculatedAt, Integer numberOfFiles,
+                STATUS currentStatus, String jobId)
+        {
+            this.id = id;
+            this.sizeInBytes = sizeInBytes;
+            this.calculatedAt = calculatedAt;
+            this.numberOfFiles = numberOfFiles;
+            this.status = currentStatus;
+            this.jobId = jobId;
+        }
+
+        public static NodeSizeDetails parseNodeSizeDetails(JSONObject jsonObject)
```

Comment:
This method is used only in tests. It would be good to move it to the test code.

Reply:
We have incorporated integration test cases in REST API TAS tests part 2.
Thus, we don't need this parse method, and we removed the same.